diff --git a/node_modules/gatsby-source-contentful/normalize.js b/node_modules/gatsby-source-contentful/normalize.js
index d3a17c4..953eae5 100644
--- a/node_modules/gatsby-source-contentful/normalize.js
+++ b/node_modules/gatsby-source-contentful/normalize.js
@@ -199,6 +199,7 @@ function prepareTextNode(node, key, text, createNodeId) {
 }
 
 function prepareRichTextNode(node, key, content, createNodeId) {
+  richTextOneLevel(content);
   const str = stringify(content);
   const richTextNode = Object.assign({}, content, {
     id: createNodeId(`${node.id}${key}RichTextNode`),
@@ -469,3 +470,34 @@ exports.createAssetNodes = ({
     createNode(assetNode);
   });
 };
+
+function richTextOneLevel(content) {
+  Object.keys(content).forEach(k => {
+    if (Array.isArray(content[k])) {
+      Object.keys(content[k]).forEach(ik => {
+        const inside = content[k][ik];
+
+        if (Object.prototype.hasOwnProperty.call(inside, 'content') && Array.isArray(inside.content)) {
+          inside.content.forEach((node, i) => {
+            if (Object.prototype.hasOwnProperty.call(node, 'nodeType') && node.nodeType === 'entry-hyperlink') {
+              // fields needed just slug for now
+              const targetFields = node.data.target.fields;
+
+              if (targetFields && Object.prototype.hasOwnProperty.call(targetFields, 'slug')) {
+                const slug = targetFields.slug;
+                node.data.target.fields = {
+                  slug
+                };
+              } else {
+                // If no slug lets delete so no max loop happens
+                if(inside.content) {
+                    inside.content.splice(i, 1);
+                }
+              }
+            }
+          });
+        }
+      });
+    }
+  })
+}
